### PR TITLE
Feat: added watch button on the slides

### PR DESCRIPTION
**Pull request by nischal-shetty2**
fixes: #233
Added the Watch button which takes us to the slide's related video

https://github.com/code100x/daily-code/assets/136328410/f4911a82-45fe-4acd-8acf-8a5d55ba3b42

### CODE REVIEW

**Comment by nischal-shetty2** [comment]
@hkirat if this pr is needed then ill add all the remaining seed links

**Comment by nischal-shetty2** [comment]
@hkirat added all the links anyway! now its all done

**Comment by Vivek-1102** [comment]
Its great feature but there may be a problem as the lectures are only for cohort people and slides are available for all so how watch button will act for those people which are not part of the cohort?

**Comment by nischal-shetty2** [comment]
> Its great feature but there may be a problem as the lectures are only for cohort people and slides are available for all so how watch button will act for those people which are not part of the cohort?

app.100xdevs wont allow you to watch it and will ask you to log in, so only cohort people can access it, and some slides are from youtube which anyone can watch

**Comment by Vivek-1102** [comment]
Then it's great feature to add

**Comment by nimit9** [comment]
Someone will have to manually add the links in the prod database.

The seedsData only works for local development.

**Comment by nischal-shetty2** [comment]
> Someone will have to manually add the links in the prod database.
> 
> The seedsData only works for local development.

how different is the production seed data? it must be pretty similar right or the same thing?

**Comment by nimit9** [comment]
There is no prod seed data.

Seeding is only for local development, it's used to populate data in your local db to make it work on local.

**Comment by nischal-shetty2** [comment]
> There is no prod seed data.
> 
> Seeding is only for local development, it's used to populate data in your local db to make it work on local.

oh then we do it manually but its still just copy pasting

**Comment by nimit9** [comment]
The prod db already has the data in the postgres db.

Right now someone manually adds the data for a new track and problem by writing an SQL query.

Someone has improved the admin dashboard, by making a form on daily code for admins to add a new track and problems. The PR is in review ig.

So someone will have to manually write SQL queries to add the links to previous all tracks. 

It comes with a lot of overhead, so not really sure if it's needed.

**Comment by nischal-shetty2** [comment]
@nimit9 got it! but its a one time thing and pretty helpful for the future since we still have 1-100 left because the faster you add it the less overhead if its needed in the future, lets see what @hkirat says

File: packages/ui/src/BlogAppbar.tsx [hunk 1]
```diff
@@ -59,6 +58,15 @@ export const BlogAppbar = ({ problem, track }: { problem: Problem; track: Track
         </div>
 
         <div className="flex space-x-2">
+          <Link target="_blank" href={`${track.link}`}>
```
Comment (hkirat) [inline packages/ui/src/BlogAppbar.tsx:61]: Please render this conditionally only if the link exists

**Comment by hkirat** [comment]
I agree with @nimit9 dont think this is needed

**Comment by nischal-shetty2** [comment]
@hkirat  this pr #176 will make it very easy to add links though

**Pull request closed without merging**